rules/code-quality/src/Rector/For_/ForToForeachRector.php

```diff
@@ -132,14 +138,26 @@ public function refactor(Node $node): ?Node
             return null;
         }
 
-        $iteratedVariableSingle = Inflector::singularize($iteratedVariable);
+        $iteratedVariableSingle = self::inflector()->singularize($iteratedVariable);
         $foreach = $this->createForeach($node, $iteratedVariableSingle);
 
         $this->useForeachVariableInStmts($foreach->expr, $foreach->valueVar, $foreach->stmts);
 
         return $foreach;
     }
 
+    /**
+     * Get the inflector instance.
+     */
+    private static function inflector(): Inflector
+    {
+        if (self::$inflector === null) {
+            self::$inflector = InflectorFactory::create()->build();
+        }
+
+        return self::$inflector;
+    }
```

Comment:
We use constructor injection here. The service must be registered in config.php

Reply:
Fixed